library/camel-kamelets-utils/src/main/java/org/apache/camel/kamelets/utils/transform/ExtractField.java

```diff
@@ -27,11 +27,16 @@
 
 public class ExtractField {
 
-    public void process(@ExchangeProperty("field") String field, Exchange ex) {
+    public void process(@ExchangeProperty("field") String field, @ExchangeProperty("headerOutput") boolean headerOutput, Exchange ex) {
+        final String EXTRACTED_FIELD_HEADER = "CamelKameletsExtractFieldName";
         ObjectMapper mapper = new ObjectMapper();
         JsonNode jsonNodeBody = ex.getMessage().getBody(JsonNode.class);
         Map<Object, Object> body = mapper.convertValue(jsonNodeBody, new TypeReference<Map<Object, Object>>(){});
-        ex.getMessage().setBody(body.get(field));
+        if (!headerOutput) {
+            ex.getMessage().setBody(body.get(field));
+        } else {
+            ex.getMessage().setHeader(EXTRACTED_FIELD_HEADER, body.get(field));
```

Comment:
As per my comments in the issue, maybe this is better in line 38?

ex.getMessage.setHeader(sanitized(headerOutput), body.get(field));

Where ```String sanitized(String headerOutput)``` does its best to ensure a valid headerName that does not collide with existing headers.

Reply:
Header output is a boolean. The header name is static and fixed. So, no.

Reply:
Also this is a bean that should be simple as simple. I troducing more logic to check stuff like headers existence is too much, there is for example the has header action or you can write your own kamelet action doing all the logic you want. But I'd probably leave the beans simplest as possible.